src/sdk/namespace_cache.js

```diff
@@ -259,15 +264,30 @@ class NamespaceCache {
                             xattr: object_info_hub.xattr,
                             last_modified_time: (new Date(object_info_hub.create_time)).getTime(),
                         };
+
+                        const start = process.hrtime.bigint();
                         const upload_res = await this.namespace_nb.upload_object(upload_params, object_sdk);
-                        this.stats_collector.update_namespace_cache_stats({
+
+                        // update latency stats on 'end'
+                        upload_res.once('end', () => {
```

Comment:
not relevant to upload calls, only read_object_stream.
so after `await upload_object` we can just update the stats.